src/App.tsx

```diff
@@ -1,39 +1,108 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import Carousel from './components/Carousel';
 
-interface State {
-  images: string[];
-}
+const images = [
+  './img/1.png',
+  './img/2.png',
+  './img/3.png',
+  './img/4.png',
+  './img/5.png',
+  './img/6.png',
+  './img/7.png',
+  './img/8.png',
+  './img/9.png',
+  './img/10.png',
+];
 
-class App extends React.Component<{}, State> {
-  state = {
-    images: [
-      './img/1.png',
-      './img/2.png',
-      './img/3.png',
-      './img/4.png',
-      './img/5.png',
-      './img/6.png',
-      './img/7.png',
-      './img/8.png',
-      './img/9.png',
-      './img/10.png',
-    ],
-  };
+const App: React.FC = () => {
+  const [itemWidth, setItemWidth] = useState(130);
+  const [size, setSize] = useState(3);
+  const [step, setStep] = useState(3);
+  const [duration, setDuration] = useState(1000);
+  const [infinite, setInfinite] = useState(false);
```

Comment:
It's not advisable to generate such a large number of states when we only have one instance (settings). 
It'd be much better to utilize a single state with an object inside.

1. Create a variable outside the Component
```
const initialSettingsState = {
  itemWidth: 130,
  size: 3,
  step: 3,
  duration: 1000,
  infinite: false
}
```

2. Use one state for settings
3. Whenever you need to change value of any field you should pass two arguments `(key, value)` and then just change a value of a settings object by key. In this case you will reuse one handler instead of handlers for each input
Example: 

```
const handleInputChange = (key, value) => {
  setSettings(prevInputs => ({...prevSettings, [key]: value}))
}
```


```suggestion
  const [settings, setSettings] = useState(initialSettingsState);
```